packages/mui-system/src/cssVars/useCurrentColorScheme.test.js

```diff
@@ -85,25 +85,23 @@ describe('useCurrentColorScheme', () => {
   });
 
   it('trigger a re-render for a multi color schemes', () => {
+    let effectRunCount = 0;
     function Data() {
       const { mode } = useCurrentColorScheme({
         supportedColorSchemes: ['light', 'dark'],
         defaultLightColorScheme: 'light',
         defaultDarkColorScheme: 'dark',
       });
-      const count = React.useRef(0);
       React.useEffect(() => {
-        count.current += 1;
+        effectRunCount += 1;
       });
-      return (
-        <div>
-          {mode}:{count.current}
-        </div>
-      );
+      return <div>{mode}</div>;
     }
+
     const { container } = render(<Data />);
 
-    expect(container.firstChild.textContent).to.equal('light:2'); // 2 because of double render within strict mode
+    expect(container.firstChild.textContent).to.equal('light');
+    expect(effectRunCount).to.equal(reactMajor >= 19 ? 2 : 3);
```

Comment:
This test was [recently added](https://github.com/mui/material-ui/pull/44451/files#diff-944bf40a71cc84d2fa1fc099090dd701aafcb0d2abfe2f3173e169b9c5b38b5b), that's why we didn't cover it before in this PR.

The previous approach didn't work with React 19, probably due to the changes to strict mode. The difference in events between 18 and 19 is the following:

```diff
 render
 render
 run effect
-run effect (this one doesn't run in React 19)
 render
 render
 run effect
```

And for both, the last effect run wasn't being registered in the rendered content. That's why with React 18 the result was `light:2` even though the effect runs 3 times. For React 19, the effect is runs 2 times.

Reply:
cc: @siriwatknp